flink-libraries/flink-table/src/test/java/org/apache/flink/api/java/io/RowCsvInputFormatTest.java

```diff
@@ -0,0 +1,1075 @@
+			assertNotNull(result);
+			assertEquals("abc", result.productElement(0));
+			assertEquals("", result.productElement(1));
+			assertEquals("hhg", result.productElement(2));
+
+			result = format.nextRecord(result);
+			assertNotNull(result);
+			assertEquals("", result.productElement(0));
+			assertEquals("", result.productElement(1));
+			assertEquals("", result.productElement(2));
+
+			result = format.nextRecord(result);
+			assertNull(result);
+			assertTrue(format.reachedEnd());
+		}
+		catch (Exception ex) {
+			fail("Test failed due to a " + ex.getClass().getName() + ": " + ex.getMessage());
+		}
+	}
+
+	@Test
+	public void testIntegerFields() throws IOException {
+		try {
+			String fileContent = "111|222|333|444|555\n666|777|888|999|000|\n";
+			FileInputSplit split = createTempFile(fileContent);
+
+			RowTypeInfo typeInfo = new RowTypeInfo(
+					new TypeInformation<?>[] {
+						BasicTypeInfo.INT_TYPE_INFO,
+						BasicTypeInfo.INT_TYPE_INFO,
+						BasicTypeInfo.INT_TYPE_INFO,
+						BasicTypeInfo.INT_TYPE_INFO,
+						BasicTypeInfo.INT_TYPE_INFO
+					});
+			CsvInputFormat<Row> format = new RowCsvInputFormat(PATH, "\n", "|", typeInfo);
+
+			format.setFieldDelimiter("|");
+
+			format.configure(new Configuration());
+			format.open(split);
+
+			Row result = new Row(5);
+
+			result = format.nextRecord(result);
+			assertNotNull(result);
+			assertEquals(Integer.valueOf(111), result.productElement(0));
+			assertEquals(Integer.valueOf(222), result.productElement(1));
+			assertEquals(Integer.valueOf(333), result.productElement(2));
+			assertEquals(Integer.valueOf(444), result.productElement(3));
+			assertEquals(Integer.valueOf(555), result.productElement(4));
+
+			result = format.nextRecord(result);
+			assertNotNull(result);
+			assertEquals(Integer.valueOf(666), result.productElement(0));
+			assertEquals(Integer.valueOf(777), result.productElement(1));
+			assertEquals(Integer.valueOf(888), result.productElement(2));
+			assertEquals(Integer.valueOf(999), result.productElement(3));
+			assertEquals(Integer.valueOf(000), result.productElement(4));
+
+			result = format.nextRecord(result);
+			assertNull(result);
+			assertTrue(format.reachedEnd());
+		}
+		catch (Exception ex) {
+			fail("Test failed due to a " + ex.getClass().getName() + ": " + ex.getMessage());
+		}
+	}
+
+	@Test
+	public void testEmptyFields() throws IOException {
+		try{
+			String fileContent = 
+					"|0|0|0|0|0|\n" +
+					"1||1|1|1|1|\n" +
+					"2|2||2|2|2|\n" +
+					"3|3|3||3|3|\n" +
+					"4|4|4|4||4|\n" +
+					"5|5|5|5|5||\n";
+			
+			FileInputSplit split = createTempFile(fileContent);
+			
+			//TODO: FLOAT_TYPE_INFO and DOUBLE_TYPE_INFO don't handle correctly null values
+			RowTypeInfo typeInfo = new RowTypeInfo(new TypeInformation<?>[] {
+				BasicTypeInfo.SHORT_TYPE_INFO,
+				BasicTypeInfo.INT_TYPE_INFO,
+				BasicTypeInfo.LONG_TYPE_INFO,
+				BasicTypeInfo.INT_TYPE_INFO,
+//				BasicTypeInfo.FLOAT_TYPE_INFO,
```

Comment:
Actually this is a problem of those parser and since the code in those functions is quite complicated I think it will be better if who wrote those parsers could have a look at them